docs/DeveloperGuide.adoc

```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
++
+Use case ends.
+
+* 4a. There are no existing archives
++
+[none]
+** 4a1.  Financial Tracker shows an empty list.
++
+Use case ends.
+
+* 5a. The given record index or archive name is invalid
++
+[none]
+** 5a1.  Financial Tracker shows an error message.
++
+Use case resumes at step 4.
+
+
+[Discrete]
+=== UC04: Deleting an archived expense
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to list all expenses under a specific archive
+2. Financial Tracker shows a list of expenses under the specific archive
+3. User requests to delete a specific expense from the archive
+4. Financial Tracker deletes the specific  expense from the specific  archive.
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 2a. The archive has no expenses in it
++
+[none]
+** 2a1. Financial tracker displays an empty list
++
+Use case ends.
+
+* 2b. The archive name is invalid
++
+[none]
+** 2b1.  Financial Tracker shows an error.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. The given record index or archive name is invalid
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1.  Financial Tracker shows an error message.
 +
 Use case resumes at step 2.
 
+
+[Discrete]
+=== UC05: Deleting an entire archive
+
+*MSS*
+
+1. User requests to list all existing archives
+2. Financial Tracker shows a list of all existing archives
+3. User requests to delete a specific archive from the list
+4. Financial Tracker deletes the specific archive
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no existing archives
++
+[none]
+** 2a1. Financial tracker displays an empty list
++
+Use case ends.
+
+* 3a.  The given archive name is invalid
++
+[none]
+** 3a1.  Financial Tracker shows an error.
++
+Use case resumes at step 2.
+
+
+[Discrete]
+=== UC06: Finding a record with a specified keyword
+
+*MSS*
+
+1. User enters a keyword to search for records containing it
+2. Financial Tracker displays a list of records that contains the specified keyword
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no records containing the specified keyword
++
+[none]
+** 2a1. Financial tracker displays an empty list
++
+Use case ends.
+
+
 _{More to be added}_
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 expenses/income records without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```

Comment:
Maybe can refine this NFR by indicating how typing is faster than using mouse? Quantitatively?